### ISSUE TITLE
[AKS] Make Azure cloud credential Tenant ID optional everywhere

### ISSUE THREAD

**Issue by cmurphy** [rancher/rancher#33097]
**What kind of request is this (question/bug/enhancement/feature request):** enhancement

Following up on https://github.com/rancher/rancher/issues/32897 the Tenant ID field should not have a red * when creating an Azure cloud credential, as it's now marked optional in the API.

![image](https://user-images.githubusercontent.com/1220710/121258000-35829580-c863-11eb-955a-753da7c274e1.png)

**Comment by westlywright** [comment]
Since the tenant id is required by the AKS v2 stuff I will also be adding a block of help text informing the user that cloud creds without a tenant id will not be visible when launching a hosted AKS cluster. When launching a new V2 AKS cluster the UI will filter all cc's without a tenant id from the selection dropdown.

**Comment by prabalsharma** [comment]
Implementation is working as commented by Wes.

**Comment by prabalsharma** [comment]
Cloud credential without Tenant ID is not getting filtered out from the selection dropdown, for imports.
This results in error: 
`Failed to load Clusters from Azure: undefined`

**Comment by gaktive** [comment]
2 PRs should sort this out in the backend: https://github.com/rancher/machine/pull/138 and https://github.com/rancher/rancher/pull/33680

For the UI, we should be remove the filtering around the tenant ID.

If errors appear from the backend, we can release note it.

**Comment by cmurphy** [comment]
Merged backend components

**Comment by cmurphy** [comment]
New revisions on backend components are merged

**Comment by snasovich** [comment]
@gaktive @nwmac , please let us know when UI changes are done so we can move this to test. Thank you.

**Comment by gaktive** [comment]
From what I understand, @vincent99's UI changes are already there so we can verify that the new behaviour works and then move this to QA.

**Comment by Jono-SUSE-Rancher** [comment]
@gaktive Sounds good. Let's get this validated and then placed back into test for the QA team as this was one of the final blocker issues on the AKS feature.

**Comment by gaktive** [comment]
UI completed and verified. Ready for QA.

**Comment by prabalsharma** [comment]
tested on  v2.6-6c14c339a699af2c88d1c5ea075f8eedce07f55b-head

Create CC and then use it to provision AKS v2 cluster:
```
Error creating resource group [Prabal-rg] with message azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to https://management.azure.com/subscriptions/***/resourcegroups/Prabal-rg?api-version=2019-10-01: StatusCode=404 -- Original Error: adal: Refresh request failed. Status Code = '404'. Response body: Endpoint https://login.microsoftonline.com/oauth2/token?api-version=1.0
```

Create CC while provisioning AKS v2 cluster:

<img width="1264" alt="Screen Shot 2021-08-03 at 12 05 28 PM" src="https://user-images.githubusercontent.com/3749546/128071805-f852e9b4-f823-4c40-9c6d-30128710da6a.png">

**Comment by cmurphy** [comment]
For the first problem - https://github.com/rancher/charts/pull/1392

For the second problem, it looks like the Ember UI is still requiring the tenant ID, and the Vue UI is not requiring it but sending an empty string to check the credentials.

**Comment by cmurphy** [comment]
Removing the release-note label, the end result of this issue is that even though improvements and changes were made to azure cloud credentials, there's no user-facing change in the UI.

**Comment by prabalsharma** [comment]
master-61339fb7f8ff0b7833a80cbb3913f152477871b7-head: was able to provision aks cluster. https://github.com/rancher/rancher/issues/33097#issuecomment-892093779 has been fixed on master

v2.6-7b9a232ac5abb274309a57ae1eeafdeebd5622ee-head:  was able to provision aks cluster. https://github.com/rancher/rancher/issues/33097#issuecomment-892093779 has been fixed on 2.6-head

2.5.9 to 2.6-head upgrades: 
- was able to create new AKS cluster on 2.6 using CC added on 2.5.9
- was able to edit AKSv1 cluster by providing Azure Client Secret
